### PR TITLE
Prepare private ID set for merge - script

### PR DESCRIPTION
**Pull request by ChanochShayner**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Prepare private ID set for merge - script.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Tests/Marketplace/Tests/prepare_private_id_set_for_merge_test.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    - remove the old new pack's data from private ID set
+    Then
+    - ensure that the private ID set not contain the old new pack's data
+    - ensure that in case there is no pack name no error returns, and the ID set remains as it is
+    """
+    private_id_set = remove_old_pack_from_private_id_set(PRIVATE_ID_SET, 'Workday')
+    assert WORKDAY_PACK not in private_id_set['integrations']
+
+    private_id_set = remove_old_pack_from_private_id_set(PRIVATE_ID_SET, 'Accessdata')
+    assert ACCESSDATA_PACK not in private_id_set['integrations']
```
Comment (guykeller) [inline Tests/Marketplace/Tests/prepare_private_id_set_for_merge_test.py:60]: aren't lines 57 and 60 testing the same thing?

File: Tests/Marketplace/prepare_private_id_set_for_merge.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    """
+    for entity, entity_list in private_id_set.items():
+        for item in entity_list[:]:
+            item_value = item.get(list(item.keys())[0], {})
+            if item_value.get('pack') == new_pack_name:
+                entity_list.remove(item)
+    return private_id_set
+
+
+def prepare_private_id_set_for_merge(private_id_set_path, new_pack_name):
```
Comment (guykeller) [inline Tests/Marketplace/prepare_private_id_set_for_merge.py:23]: what does this name mean?

File: Tests/Marketplace/prepare_private_id_set_for_merge.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    Args:
+        private_id_set (dict): The private ID set
+        new_pack_name (str): The name of the new pack
+
+    Returns:
+        Private id set without the old data of the new package
+    """
+    for entity, entity_list in private_id_set.items():
+        for item in entity_list[:]:
+            item_value = item.get(list(item.keys())[0], {})
```
Comment (guykeller) [inline Tests/Marketplace/prepare_private_id_set_for_merge.py:17]: names here aren't very clear as well

**Review by guykeller — CHANGES_REQUESTED** [review]

File: Tests/Marketplace/Tests/prepare_private_id_set_for_merge_test.py (hunk 1 above)
Comment (ChanochShayner) [reply Tests/Marketplace/Tests/prepare_private_id_set_for_merge_test.py:60]: Yes, but for different packs

File: Tests/Marketplace/prepare_private_id_set_for_merge.py (hunk 2 above)
Comment (ChanochShayner) [reply Tests/Marketplace/prepare_private_id_set_for_merge.py:23]: done

File: Tests/Marketplace/prepare_private_id_set_for_merge.py (hunk 3 above)
Comment (ChanochShayner) [reply Tests/Marketplace/prepare_private_id_set_for_merge.py]: done

**Review by guykeller — APPROVED** [review]

**Pull request merged**